include/validator/validator_component.h

```diff
@@ -194,6 +279,7 @@ struct SectionVisitor {
   }
   Expect<void> operator()(const ComponentSection &Sec) {
     auto &C = Sec.getContent();
+    Ctx.addComponent(C);
     V.validate(C);
```

Comment:
Maybe reverse the order more make sense? Check the component first, then use it in further validation.